crates/precompiles/src/tip20/mod.rs

```diff
@@ -2823,4 +2823,106 @@ pub(crate) mod tests {
             Ok(())
         })
     }
+
+    #[test]
+    fn test_change_transfer_policy_id_pre_allegro_moderato() -> eyre::Result<()> {
+        let mut storage = HashMapStorageProvider::new(1).with_spec(TempoHardfork::Allegretto);
+        let admin = Address::random();
+
+        StorageCtx::enter(&mut storage, || {
+            let mut factory = TIP20Factory::new();
+            factory.initialize()?;
+
+            // Deploy PathUSD token
+            deploy_path_usd(&mut factory, admin)?;
+            let mut token = TIP20Token::from_address(PATH_USD_ADDRESS)?;
+
+            // Before Allegro Moderato, policy validation should be skipped
+            let arbitrary_policy_id = 999u64;
+            let result = token.change_transfer_policy_id(
+                admin,
+                ITIP20::changeTransferPolicyIdCall {
+                    newPolicyId: arbitrary_policy_id,
+                },
+            );
+
+            assert!(result.is_ok());
+
```

Comment:
we could add here assertions for the value to be actually stored, like `assert!(token.transfer_policy_id(), arbitrary_policy_id);` wdyt?

Reply:
Sounds good, updated to verify `assert_eq!(token.transfer_policy_id()?, arbitrary_policy_id);` below.